File: solutions/3096. Minimum Levels to Gain More Points/3096.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def minimumLevels(self, possible: List[int]) -> int:
    bobScore = 0
    for num in possible:
      bobScore += 1 if num != 0 else -1

    danielScore = 0
    for i in range(len(possible) - 1):
      danielScore += 1 if possible[i] != 0 else -1
      if danielScore > bobScore - danielScore:
        return i + 1

    return -1    

File: solutions/3096. Minimum Levels to Gain More Points/3096.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  int minimumLevels(const vector<int>& possible) {
    int bobScore = 0;
    for (const int& num : possible)
      bobScore += (num ? 1 : -1);
        
    int danielScore = 0;
    for (int i = 0; i < possible.size() - 1; ++i) {
      danielScore += (possible[i] ? 1 : -1);
      if (danielScore > bobScore - danielScore)
        return i + 1;
    }
        
    return -1;
  }
};